frontend/src/plugin/runPlugin.ts

```diff
@@ -0,0 +1,209 @@
+  packageVersion: string,
+  /** call back when an execution error occurs in the plugin */
+  handleError: (error: unknown, packageName: string, packageVersion: string) => void,
+  /** is a non global private copy of Function */
+  PrivateFunction: typeof Function,
+  /** The argument names to be given to the plugin */
+  args: string[],
+  /** The values for the arguments to be given to the plugin */
+  values: unknown[],
+  /** A private copy of runPlugin */
+  internalRunPlugin: typeof runPlugin,
+  /** A private copy of console.error */
+  consoleError: typeof console.error
+];
+
+/**
+ * Runs a plugin by executing the source code in the global scope.
+ *
+ * This provides a way to pass private variables to individual plugins.
+ *
+ * @param source source code of plugin
+ * @param pluginPath path to plugin
+ * @param packageName name of package
+ * @param packageVersion version of package
+ * @param permissionSecrets permission secrets are keyed by permission name, valued by secret
+ * @param handleError call back when an execution error occurs in the plugin
+ * @param getAllowedPermissions call back which returns only allowed permission secrets for plugin
+ * @param getArgValues call back which returns the argument names and values for the plugin
+ * @param privateFunction is a non global private copy of Function
+ */
+export function getInfoForRunningPlugins({
+  source,
+  pluginPath,
+  packageName,
+  packageVersion,
+  permissionSecrets,
+  handleError,
+  getAllowedPermissions,
+  getArgValues,
+  PrivateFunction,
+  internalRunPlugin,
+  consoleError,
+}: {
+  source: string;
+  pluginPath: string;
+  packageName: string;
+  packageVersion: string;
+  permissionSecrets: Record<string, number>;
+  handleError: (error: unknown, packageName: string, packageVersion: string) => void;
+  getAllowedPermissions: (
+    pluginName: string,
+    pluginPath: string,
+    permissionSecrets: Record<string, number>
+  ) => Record<string, number>;
+  getArgValues: (
+    pluginName: string,
+    pluginPath: string,
+    allowedPermissions: Record<string, number>
+  ) => [string[], unknown[]];
+  PrivateFunction: typeof Function;
+  internalRunPlugin: typeof runPlugin;
+  consoleError: typeof console.error;
+}): runPluginProps | undefined {
+  if (!pluginPath || !packageName || !packageVersion) {
+    consoleError(`Either pluginPath, packageName or packageVersion is missing for ${pluginPath}`);
+    return;
+  }
+
+  const sourceMapPathForDebugging = `\n//# sourceURL=//${pluginPath}/dist/main.js`;
+  const allowedPermissions = getAllowedPermissions(packageName, pluginPath, permissionSecrets);
+  const argsValues = getArgValues(packageName, pluginPath, allowedPermissions);
+  const args = argsValues[0];
+  const values = argsValues[1];
+
+  return [
+    source + sourceMapPathForDebugging,
+    packageName,
+    packageVersion,
+    handleError,
+    PrivateFunction,
+    args,
+    values,
+    internalRunPlugin,
+    consoleError,
+  ];
```

Comment:
You probably have a good reason to process this as a list rather than an object?